parlai/core/worlds.py

```diff
@@ -359,12 +367,21 @@ def episode_done(self):
         return done
 
     def report(self, compute_time=False):
-        metrics = self.agents[0].report()
-        if compute_time:
-            self.total_exs += metrics['total']
-            time_metrics = compute_time_metrics(self, self.opt['max_train_time'])
-            metrics.update(time_metrics)
-        return metrics
+        metrics = {}
+        for a in self.agents:
+            if hasattr(a, 'report'):
+                m = a.report()
+                for k, v in m.items():
+                    if k not in metrics:
+                        # first agent gets priority in settings values for keys
+                        # this way model can't e.g. override accuracy to 100%
+                        metrics[k] = v
+        if metrics:
+            if compute_time and 'total' in metrics:
```

Comment:
are there cases when we do not count the total?

Reply:
eh not that I can think of but seems reasonable to check